lib/src/provider/asset_picker_provider.dart

```diff
@@ -254,6 +254,155 @@ abstract class AssetPickerProvider<Asset, Path> extends ChangeNotifier {
+    _autoScroller = EdgeDraggingAutoScroller(
+      scrollableState,
+      velocityScalar: _kDefaultAutoScrollVelocityScalar,
+    );
+
+    initialSelectedIdx = index;
+
+    dragSelect = !selectedAssets.contains(entity);
+  }
+
+  void onDragUpdate(
+    BuildContext context,
+    LongPressMoveUpdateDetails details,
+    double itemSize,
+    int gridCount,
+    double topPadding,
+  ) {
+    if (!isInDragging) {
+      return;
+    }
+
+    if (dragSelect && selectedAssets.length == maxAssets) {
+      return;
+    }
+
+    final scrollableState = _checkScrollableStatePresent(context);
+    if (scrollableState == null) {
+      return;
+    }
+
+    final column = _getIndex(details.globalPosition.dx, itemSize);
+    final row = _getIndex(
+      details.globalPosition.dy -
+          topPadding -
+          24.0 +
+          scrollableState.position.pixels,
+      itemSize,
+    );
+
+    final currentDragIndex = row * gridCount + column;
+
+    List<Asset> filteredAssetList = <Asset>[];
+    // add asset
+    if (currentDragIndex < initialSelectedIdx) {
+      filteredAssetList = _currentAssets
+          .getRange(
+            currentDragIndex,
+            math.min(initialSelectedIdx + 1, currentAssets.length),
+          )
+          .toList()
+        ..reversed;
+    } else {
+      filteredAssetList = _currentAssets
+          .getRange(
+            initialSelectedIdx,
+            math.min(currentDragIndex + 1, currentAssets.length),
+          )
+          .toList();
+    }
+
+    filteredAssetList.forEach(dragSelect ? selectAsset : unSelectAsset);
+
+    _autoScroller?.startAutoScrollIfNecessary(
+      Rect.fromLTWH(
+        (column + 1) * itemSize,
+        details.globalPosition.dy > MediaQuery.sizeOf(context).height * 0.8
+            ? (row + 1) * itemSize
+            : math.max(topPadding, details.globalPosition.dy),
+        itemSize,
+        itemSize,
+      ),
+    );
+  }
+
+  void onDragEnd(LongPressEndDetails details) {
+    resetDraggingStatus();
+  }
+
+  /// 复原拖拽状态
+  /// Reset dragging status
+  void resetDraggingStatus() {
+    isInDragging = false;
+    initialSelectedIdx = -1;
+    dragSelect = true;
+    _autoScroller?.stopAutoScroll();
+    _autoScroller = null;
+  }
+
+  /// 检查[Scrollable] state是否存在
+  ///
+  ScrollableState? _checkScrollableStatePresent(BuildContext context) {
+    final scrollable = Scrollable.maybeOf(context);
+    assert(
+      scrollable != null,
+      'To use drag and select function, Scrollable state must be the present to get the actual item position.',
+    );
+    assert(
+      scrollable?.position.axis == Axis.vertical,
+      'To use drag and select function. The Scrollable Axis must be in vertical direction',
+    );
+
+    if (scrollable == null || scrollable.position.axis != Axis.vertical) {
+      resetDraggingStatus();
+      return null;
+    }
+
+    return scrollable;
+  }
+
+  /// 获取坐标
+  /// Get Coordinate Helper
+  int _getIndex(double delta, double itemSize) {
```

Comment:
What index is it trying to get? Please make a more clear statement such as `_getSelectingAssetIndex`.

Reply:
I have updated the naming to provide a clearer statement.